jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -127,18 +127,18 @@ def _on_glue_y_display_unit_changed(self, y_unit):
 
     @observe('spectral_unit_selected')
     def _on_spectral_unit_changed(self, *args):
-        self.hub.broadcast(GlobalDisplayUnitChanged('spectral',
-                                                    self.spectral_unit.selected,
-                                                    sender=self))
         xunit = _valid_glue_display_unit(self.spectral_unit.selected, self.spectrum_viewer, 'x')
         if self.spectrum_viewer.state.x_display_unit != xunit:
             self.spectrum_viewer.state.x_display_unit = xunit
+        self.hub.broadcast(GlobalDisplayUnitChanged('spectral',
```

Comment:
Can these messages actually go inside the `if` block? We don't need to broadcast this if the unit didn't actually change, do we?

Reply:
What was the reason for changing the order?  I'm trying to think if there are any (unintentional) consequences to that... but hopefully it is fine if its needed for something else.

Reply:
I was using the `spectrum_viewer` to get the spectral units while developing so I moved it to after that change. Now that it is in the if statement I don't know if it matters anymore.